python/cuml/cuml/linear_model/logistic_regression.pyx

```diff
@@ -282,6 +241,45 @@ class LogisticRegression(UniversalBase,
         Fit the model with X and y.
 
         """
+        if self.penalty not in supported_penalties:
+            raise ValueError("`penalty` " + str(self.penalty) + " not supported.")
+
+        if self.solver not in supported_solvers:
+            raise ValueError("Only quasi-newton `qn` solver is "
+                             " supported, not %s" % self.solver)
+
+        if self.penalty == "none":
+            warnings.warn(
+                "The 'none' option was deprecated in version 24.06, and will "
+                "be removed in 25.08. Use None instead.",
+                FutureWarning
+            )
+            penalty = None
+        else:
+            penalty = self.penalty
+
+        if penalty == "elasticnet":
+            if self.l1_ratio is None:
+                raise ValueError(
+                    "l1_ratio has to be specified for" "loss='elasticnet'"
+                )
+            if self.l1_ratio < 0.0 or self.l1_ratio > 1.0:
+                msg = "l1_ratio value has to be between 0.0 and 1.0"
+                raise ValueError(msg.format(self.l1_ratio))
```

Comment:
I wonder if we should have a function in the estimators, something like `self._validate_params` or something, so that fit/predict functions stay slim and easy to follow, and this logic is isolated and we can add unit tests?

Reply:
Good idea.

---

For the more general idea of parameter validation: scikit-learn uses a hybrid approach. There is a parameter validation "framework" https://github.com/scikit-learn/scikit-learn/blob/856ef1f72359d17bb3687b8accbf7c11eebdc947/sklearn/utils/_param_validation.py#L28 that takes care of basics like "is this parameter an int and between 0 and 100?". And then we have Python code that takes care of more involved validation or rules where one parameter's legal values depend on the value of some other parameter

Probably quite a bit of work to migrate to a system like this, so for now just "for our interest"